edb/schema/constraints.py

```diff
@@ -825,6 +827,27 @@ def _populate_concrete_constraint_attrs(
                 context=expr_context
             )
 
+        if (subjectexpr is not None and isinstance(subject_obj, s_types.Type)
+                and subject_obj.is_object_type()):
+            final_subjectexpr = s_expr.Expression.compiled(
+                subjectexpr,
+                schema=schema,
+                options=qlcompiler.CompilerOptions(
+                    anchors={qlast.Subject().name: subject},
+                    singletons=frozenset({subject_obj}),
+                ),
+            )
+            assert isinstance(final_subjectexpr.irast, ir_ast.Statement)
+
+            if final_subjectexpr.irast.cardinality.is_multi():
+                refs = ir_utils.get_longest_paths(final_expr.irast)
+                if len(refs) > 1:
+                    raise errors.InvalidConstraintDefinitionError(
+                        "Constraint with multi cardinality may not "
+                        "reference multiple fields",
```

Comment:
We use "pointers" instead of "fields" (and that's how the internal abstraction is called).  That said, we also go out of our way to _not_ use that term in the documentation and instead use "link or property". @edgedb/engineering, interested to hear opinions on whether we want a single term to mean "link or property", and what should it be.

Reply:
"Link or property" is fine. Single terms tend to be confusing if they're not obvious. Ideally we should use either "link" or "property" in error messages when we know which one is more relevant.